vktrace/vktrace_common/vktrace_trace_packet_utils.c

```diff
@@ -113,16 +114,163 @@ uint64_t vktrace_get_time() {
 uint64_t vktrace_get_time() { return 0; }
 #endif
 
+uint64_t get_endianess() {
+    uint64_t x = 1;
+    return *((char*)&x) ? VKTRACE_BIG_ENDIAN : VKTRACE_LITTLE_ENDIAN;
+}
+
+uint64_t get_arch() {
+    uint64_t rval;
+#if defined(PLATFORM_LINUX)
+    struct utsname buf;
+    uname(&buf);
+    strncpy((char*)&rval, buf.machine, sizeof(uint64_t));
+#else
+    SYSTEM_INFO systemInfo;
+    char* arch;
+    GetSystemInfo(&systemInfo);
+    switch (systemInfo.wProcessorArchitecture) {
+        case PROCESSOR_ARCHITECTURE_AMD64:
+            arch = "x86_64";
+            break;
+        case PROCESSOR_ARCHITECTURE_ARM:
+            arch = "ARM";
+            break;
+        case PROCESSOR_ARCHITECTURE_IA64:
+            arch = "ia64";
+            break;
+        case PROCESSOR_ARCHITECTURE_INTEL:
+            arch = "x86";
+            break;
+        default:
+            arch = "";
+            break;
+    }
+    strncpy((char*)&rval, arch, sizeof(uint64_t));
+#endif
+    return rval;
+}
+
+uint64_t get_os() {
+    uint64_t rval;
+#if defined(ANDROID)
+    strncpy((char*)&rval, "Android", sizeof(uint64_t));
+#elif defined(PLATFORM_LINUX)
+    strncpy((char*)&rval, "Linux", sizeof(uint64_t));
+#else
+    strncpy((char*)&rval, "Windows", sizeof(uint64_t));
+#endif
+    return rval;
+}
+
+static VkPhysicalDeviceProperties* get_physicalDeviceProperties() {
+    VkPhysicalDeviceProperties* rval = NULL;
+
+    VkInstanceCreateInfo createInfo;
+    VkInstance instance;
+    uint32_t physDevCount;
+    VkPhysicalDevice physDevice;
+
+    static bool devPropertiesValid = false;
+    static VkPhysicalDeviceProperties devProperties;
+
+    if (devPropertiesValid) {
+        rval = &devProperties;
+        goto allDone;
```

Comment:
this goto allDone doing nothing but returns, we can return right here as well and use less goto.